### PR TITLE
Feature/mutations queries

### PR DESCRIPTION
**Pull request by kevinhartmann23**
## *Purpose:*
*What does this merge do? Check all that apply.*
- [x] Adds a new feature
- [x] Improves an existing feature
- [ ] Fixes a bug
- [ ] Cleans up (logs, tests, etc.)

## *Description: What does this PR do?*
*Briefly, but clearly describe what you did, and explain any new code.*
Adds successful queries, and start to shells of mutations
MUTATIONS NEED A BIT OF STUDYING WITH useMutation hook

## *Instructions for reviewer:*
*Where should the reviewer start?*
*How should this be manually tested?*
*Any background context you want to provide?*
*What are the relevant tickets?*

## *Issues:*
*Are there any concerns, issues, or bugs in this branch? If so describe them.*

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have used tests to prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my change

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@apollo/client": "^3.3.15",
```
Comment (kristenmb) [inline package.json:6]: Important to note for everyone: We need to install apollo client and remove other apollo related packages.

File: src/components/form/Form.js [hunk 2]
```diff
@@ -48,10 +80,93 @@ const Form = () => {
...
-          incrementForm={incrementForm}
-        />)}
-        { formCounter > 1 && <FormTime incrementForm={incrementForm}/> }
+        { formCounter > 0 && (
+          <section className='form-section'>
+            <select value={selectedVenue} onChange={(event) => handleVenueSelect(event)}>
+              <option value={''}>Please Select A Venue</option>
+              <option value={'newVenue'}>Add New Venue</option>
+              {findVenues()}
+            </select>
```
Comment (kristenmb) [inline src/components/form/Form.js:89]: Im glad we finally sorted this out! That was a confusing challenge.

**Review by kristenmb — APPROVED** [review]
The queries are looking great and I know we're super close with the mutations. Lets keep digging into the docs. We'll sort it out I'm sure.

File: package.json (hunk 1 above)
Comment (DrewBradley) [reply package.json:6]: Thanks for the heads up!

File: src/components/App.js [hunk 3]
```diff
@@ -8,19 +8,12 @@ import Form from './form/Form'
...
 
 
 function App() {
-  const eventsData = useQuery(EVENTS_QUERY)
-
-  const {setEvents} = useApp()
-
-  useEffect(() => {
-    setEvents(eventsData.data)
-  },[eventsData])
```
Comment (DrewBradley) [inline src/components/App.js:17]: Thanks for letting me try this out! I'm glad you got onComplete figured out!

File: src/components/dashboard/TourDisplay.js [hunk 4]
```diff
@@ -1,11 +1,18 @@
 import React from 'react'
 import './Dashboard.css'
+import {useApp} from '../../context/AppContext'
+
+const TourDisplay = () => {
+  const {tours} = useApp()
```
Comment (DrewBradley) [inline src/components/dashboard/TourDisplay.js:6]: Heck yes! Global state for the win!

File: src/components/form/Form.js [hunk 5]
```diff
@@ -1,25 +1,57 @@
...
+
+  const validateForm = (event) => {
+    event.preventDefault()
+
+    const newEvent = {
+      "tourId": 1,
+      "name": eventName,
+      "venueId": selectedVenue,
+      startTime,
+      endTime
```
Comment (DrewBradley) [inline src/components/form/Form.js:49]: Does this confirmation get displayed before being sent to update the state?

File: src/components/form/Form.js (hunk 2 above)
Comment (DrewBradley) [reply src/components/form/Form.js:89]: Wow! Y'all did some work to make this so concise!

File: src/context/AppContext.js [hunk 6]
```diff
@@ -9,37 +21,118 @@ export const useApp = () => {
...
+      setError(false)
+    }
+  })
 
-  const [contacts, setContacts] = useState([])
+  useQuery(EVENTS_QUERY, {
+    onCompleted: data => {
+      setEvents(data.events)
+    }
+  })
```
Comment (DrewBradley) [inline src/context/AppContext.js:46]: I cannot believe how tiny this code looks for how much it is doing! Big wow!

File: src/context/AppContext.js [hunk 7]
```diff
@@ -9,37 +21,118 @@ export const useApp = () => {
...
+  }
 
-  // const getEvents = () => {
-    // do API call
-    // apiCall.getEvents
-    // return what we want
-  // }
 
-  // values
+  // VALUES
```
Comment (DrewBradley) [inline src/context/AppContext.js:126]: I feel like I've seen this, but is all caps the common practice for comments? Good catch. I just had no idea.

File: src/context/mutations.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+   mutation CreateEvent($input: CreateEventInput!) {
+    createEvent(input: {tourId: 1, name: "TESTING", venueId: 3, startTime: "2021-04-10", endTime: "2021-05-01"}) {
+        tourId,
+        name,
+        venueId,
+        startTime,
+        endTime
+      }
+  }
+`
```
Comment (DrewBradley) [inline src/context/mutations.js:13]: Dang GraphQL, why you gotta be putting dolla signs in front of stuff, making me think this is jQuery or something?

**Review by DrewBradley — APPROVED** [review]
This looks awesome! I love how the mutation looks.

File: package.json (hunk 1 above)
Comment (richardltyler) [reply package.json:6]: you mean run `npm install` when we pull his down?

File: src/components/dashboard/Dashboard.js [hunk 9]
```diff
@@ -42,7 +42,7 @@ const Dashboard = () => {
                 year: 'numeric',
                 day: 'numeric',
               }}
-              events={ events && events.events ? formatEvents(events.events) : []}
+              events={ events ? formatEvents(events) : []}
```
Comment (richardltyler) [inline src/components/dashboard/Dashboard.js:45]: I love how this is getting more complicated and cleaner/more readable at the same time 😂

File: src/components/dashboard/Dashboard.js [hunk 10]
```diff
@@ -10,12 +10,12 @@ import { useApp } from '../../context/AppContext'
...
 
 const Dashboard = () => {
-  const { events } = useApp()
+  const { events, tours } = useApp()
 
   return (
     <div className="Dashboard">
       <ContactsPreview/>
-      <TourDisplay name="MVP"/>
+      <TourDisplay />
```
Comment (richardltyler) [inline src/components/dashboard/Dashboard.js:18]: global state makes this so much easier simpler

File: src/components/form/Form.js [hunk 11]
```diff
@@ -1,25 +1,57 @@
 import React, {useState} from 'react'
-import FormVenue from './FormVenue'
-import FormTime from './FormTime'
+import {useApp} from '../../context/AppContext'
 
 import './Form.css';
 
 const Form = () => {
```
Comment (richardltyler) [inline src/components/form/Form.js:6]: This form looks really intuitive after all the work y'all put in. Thanks for all of this!

File: src/context/AppContext.js [hunk 12]
```diff
@@ -1,6 +1,18 @@
 import React, {useContext, useState} from 'react'
-import { useQuery } from 'react-apollo';
-import { TOURS_QUERY, EVENTS_QUERY } from './queries'
+import { useQuery, useMutation } from '@apollo/client';
+import {
```
Comment (richardltyler) [inline src/context/AppContext.js:3]: I like how these got broken out into their own file. Nice job!

File: src/index.js [hunk 13]
```diff
@@ -3,13 +3,13 @@ import ReactDOM from 'react-dom';
...
 import AppProvider from './context/AppContext'
-import ApolloClient from 'apollo-boost';
-import { ApolloProvider } from 'react-apollo';
+import {ApolloClient, InMemoryCache} from '@apollo/client';
+import { ApolloProvider } from '@apollo/client/react';
 
 const client = new ApolloClient({
-  uri: 'https://curatour-be.herokuapp.com/graphql'
+  uri: 'https://curatour-be.herokuapp.com/graphql',
+  cache: new InMemoryCache()
```
Comment (richardltyler) [inline src/index.js:12]: Woah that looks like a big development. I'll be looking into this for sure

**Review by richardltyler — APPROVED** [review]
Y'all got a LOT done and it looks great. Thanks for getting all of this in on a saturday!!

**Pull request merged**